File: OkamiBooks/Scripts/app/controllers/editController.js
(function (global, ng) {
    'use strict';

    function editController($scope, $http, $cookies, $location) {
        $http({
            method: "GET",
            url: "Home/GetCategories"
        }).then(function mySucces(response) {
            $scope.categories = response.data;
        }, function myError(response) {
            $scope.myWelcome = [{ id: 10 }, { id: 11 }, { id: 41 }];
        });
        var myData = {
            'userName': $cookies.get('userName'),
            'accsessToken': $cookies.get('accsessToken')
        };
        $http.post('/Edit/GetChap', myData).
            success(function (data) {

                $scope.chapters = data;


            }).
            error(function () {
                $location.url('/409');
            });
        $http.post('/Edit/GetName', myData).
            success(function (data) {

                $scope.name = data;

                
            }).
            error(function () {
                $location.url('/409');
            });
        $http.post('/Edit/GetCategory', myData).
            success(function (data) {

                $scope.bookCategory = data;


            }).
            error(function () {
                $location.url('/409');
            });
        $scope.newChapterMy = '';
        $scope.text = '';
        $scope.chapterNum = '';
        $scope.addChapter = function () {
            if ($scope.newChapterMy !== '') {
                $http.post(
                        '/Edit/AddChapter', {
                            userName: $cookies.get('userName'),
                            accsessToken: $cookies.get('accsessToken'),
                            chapterName: $scope.newChapterMy,
                            text: $scope.text
                        }
                    ).
                    success(function (data) {
                        $scope.chapters = data;
                        $scope.newChapterMy = '';

                    }).
                    error(function () {
                        $location.url('/409');
                    });
            }
        }
        $scope.postName = function () {

            $http.post(
                    '/Edit/PostName', {
                        userName: $cookies.get('userName'),
                        accsessToken: $cookies.get('accsessToken'),
                        bookName: $scope.name
                    }
                ).
                success(function (data) {


                }).
                error(function () {

                });

        }
        $scope.postCategory = function () {

            $http.post(
                    '/Edit/PostCategory', {
                        userName: $cookies.get('userName'),
                        accsessToken: $cookies.get('accsessToken'),
                        bookCategory: $scope.bookCategory
                    }
                ).
                success(function (data) {


                }).
                error(function () {

                });

        }
        $scope.setChapter = function (chapter) {
            $scope.chapterNum = chapter;
            $http.post(
                        '/Edit/GetText', {
                            userName: $cookies.get('userName'),
                            accsessToken: $cookies.get('accsessToken'),
                            chapterId: $scope.chapterNum
                        }
                    ).
                    success(function (data) {
                    $scope.text = data;
                }).
                    error(function () {
                        $location.url('/409');
                    });
        }
        $scope.postText = function () {
            if ($scope.chapterNum !== '') {
                $http.post(
                        '/Edit/PostText', {
                            userName: $cookies.get('userName'),
                            accsessToken: $cookies.get('accsessToken'),
                            chapterId: $scope.chapterNum,
                            text: $scope.text
                        }
                    ).
                    success(function (data) {
                    }).
                    error(function () {
                        $location.url('/409');
                    });
            }
        }
    }

    app.controller('editController', editController);
}(window, angular));

File: OkamiBooks/Controllers/EditController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using OkamiBooks.Models;
using OkamiBooks.Models.Entities;
using WebGrease.Css.Extensions;

namespace OkamiBooks.Controllers
{
    public class EditController : Controller
    {
        // GET: Edit
        public ActionResult Index()
        {
            return View();
        }
        [HttpPost]
        public JsonResult GetChap(string userName, long accsessToken)
        {
            var chapters = new List<Chapter>();
            long bookId = 0;
            using (var context = new DatabaseContext())
            {
                context.ApplicationUsers.ForEach(x =>
                {
                    if (x.UserName == userName)
                    {
                        bookId = x.BookId;

                    }
                });
                var thisBook = new Book();
                context.Books.ForEach(x =>
                {
                    if (x.Id == bookId)
                    {
                        thisBook = x;
                    }
                });
                thisBook.GetChaptersList().ForEach(x =>
                {
                    context.Chapters.ForEach(c =>
                    {
                        if (c.Id == x)
                        {
                            chapters.Add(c);
                        }
                    });
                });
            }
            return Json(chapters, JsonRequestBehavior.AllowGet);
        }
        [HttpPost]
        public JsonResult AddChapter(string userName, long accsessToken, string chapterName)
        {
            var chapters = new List<Chapter>();
            long bookId = 0;
            using (var context = new DatabaseContext())
            {
                context.SaveChanges();
                context.ApplicationUsers.ForEach(x =>
                {
                    if (x.UserName == userName)
                    {
                        bookId = x.BookId;

                    }
                });
                var thisBook = new Book();
                context.Books.ForEach(x =>
                {
                    if (x.Id == bookId)
                    {
                        thisBook = x;
                    }
                });

                context.SaveChanges();
                var id = context.IdSaviors.ToList()[0].MaxChapter++;
                context.SaveChanges();
                context.Chapters.Add(new Chapter { Name = chapterName, LastChangeTime = DateTime.Now, Book = bookId });



                thisBook.ChaptersAdd(id);

                context.SaveChanges();
                thisBook.GetChaptersList().ForEach(x =>
                {
                    context.Chapters.ForEach(c =>
                    {
                        if (c.Id == x)
                        {
                            chapters.Add(c);
                        }
                    });
                });

            }
            return Json(chapters, JsonRequestBehavior.AllowGet);
        }
        [HttpPost]
        public JsonResult PostName(string userName, long accsessToken, string bookName)
        {
            long bookId = 0;
            using (var context = new DatabaseContext())
            {
                context.SaveChanges();
                context.ApplicationUsers.ForEach(x =>
                {
                    if (x.UserName == userName)
                    {
                        bookId = x.BookId;

                    }
                });
                context.Books.ForEach(x =>
                {
                    if (x.Id == bookId)
                    {
                        x.Name = bookName;
                    }
                });
                context.SaveChanges();
            }
            return Json(null, JsonRequestBehavior.AllowGet);
        }
        [HttpPost]
        public JsonResult GetName(string userName, long accsessToken)
        {
            long bookId = 0;
            string bookName = "";
            using (var context = new DatabaseContext())
            {
                context.SaveChanges();
                context.ApplicationUsers.ForEach(x =>
                {
                    if (x.UserName == userName)
                    {
                        bookId = x.BookId;

                    }
                });
                context.Books.ForEach(x =>
                {
                    if (x.Id == bookId)
                    {
                        bookName = x.Name;
                    }
                });
                context.SaveChanges();
            }
            return Json(bookName, JsonRequestBehavior.AllowGet);
        }
        [HttpPost]
        public JsonResult GetCategory(string userName, long accsessToken)
        {
            string bookCategory = "";
            long bookId = 0;
            using (var context = new DatabaseContext())
            {
                context.SaveChanges();
                context.ApplicationUsers.ForEach(x =>
                {
                    if (x.UserName == userName)
                    {
                        bookId = x.BookId;

                    }
                });
                context.Books.ForEach(x =>
                {
                    if (x.Id == bookId)
                    {
                         bookCategory = x.Category;
                    }
                });
                context.SaveChanges();
            }
            return Json(bookCategory, JsonRequestBehavior.AllowGet);
        }
        [HttpPost]
        public JsonResult PostCategory(string userName, long accsessToken, string bookCategory)
        {
            long bookId = 0;
            using (var context = new DatabaseContext())
            {
                context.SaveChanges();
                context.ApplicationUsers.ForEach(x =>
                {
                    if (x.UserName == userName)
                    {
                        bookId = x.BookId;

                    }
                });
                context.Books.ForEach(x =>
                {
                    if (x.Id == bookId)
                    {
                        x.Category = bookCategory;
                    }
                });
                context.SaveChanges();
            }
            return Json(null, JsonRequestBehavior.AllowGet);
        }
        [HttpPost]
        public JsonResult PostText(string userName, long accsessToken, long chapterId, string text)
        {
            using (var context = new DatabaseContext())
            {
                context.SaveChanges();
                context.Chapters.ForEach(x =>
                {
                    if (x.Id == chapterId)
                    {
                        x.Text = text;
                    }
                });
                context.SaveChanges();
            }
            return Json(null, JsonRequestBehavior.AllowGet);
        }
        [HttpPost]
        public JsonResult GetText(string userName, long accsessToken, long chapterId)
        {
            string newText = "";
            using (var context = new DatabaseContext())
            {
                context.SaveChanges();
                context.Chapters.ForEach(x =>
                {
                    if (x.Id == chapterId)
                    {
                        newText = x.Text;
                    }
                });
                context.SaveChanges();
            }
            return Json(newText, JsonRequestBehavior.AllowGet);
        }

    }
}